frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/cover/change_cover_popover_bloc.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:io';
+
+import 'package:appflowy/plugins/document/presentation/plugins/cover/change_cover_popover.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+part 'change_cover_popover_bloc.freezed.dart';
+
+class ChangeCoverPopoverBloc
+    extends Bloc<ChangeCoverPopoverEvent, ChangeCoverPopoverState> {
+  ChangeCoverPopoverBloc() : super(const ChangeCoverPopoverState.initial()) {
+    on<ChangeCoverPopoverEvent>((event, emit) async {
+      await event.map(
+        fetchPickedImagePaths:
+            (FetchPickedImagePaths fetchPickedImagePaths) async {
+          final imageNames = await _getPreviouslyPickedImagePaths();
+          emit(ChangeCoverPopoverState.loaded(imageNames));
+        },
+        deleteImage: (DeleteImage deleteImage) async {
+          final currentState = state;
+          if (currentState is Loaded) {
+            await _deleteImageInStorage(deleteImage.path);
+            final updateImageList = currentState.imageNames
+                .where((path) => path != deleteImage.path)
+                .toList();
+            await _updateImagePathsInStorage(updateImageList);
+            emit(Loaded(updateImageList));
+          }
+        },
+        clearAllImages: (ClearAllImages clearAllImages) async {
+          final currentState = state;
+          if (currentState is Loaded) {
+            for (final image in currentState.imageNames) {
+              await _deleteImageInStorage(image);
+            }
+            await _updateImagePathsInStorage([]);
+            emit(const Loaded([]));
+          }
+        },
+      );
+    });
+  }
+
+  Future<List<String>> _getPreviouslyPickedImagePaths() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
```

Comment:
I would personally prefer this being part of the BLoC as a private field that is initialized. In case there is a need for it in another event, or if the event calling this method is called again and again.

Reply:
You mean instance of shared preference as private field?

Reply:
Yes,

`late final SharedPreferences _prefs;`

Reply:
I'm having trouble initializing future as field inside bloc. Can you share code for the reference ? Thank you

Reply:
The code sample I sent is what you need to do.

Reply:
Thanks @Xazin. I learned something new.

Reply:
I have a doubt. The code is working even without using _initCompleter. Is it necessary?

Reply:
Well, it's primarily to stop an edge case to happen, if let's say for some reason the platform method that returns the SharedPreferences is blocked for some reason, and you instantiate the BLoC with an event, a method could be called where _prefs is not instantiated, it's a rare edge case, but it can happen.